### PR TITLE
In JS docs, use Note: consistently at the start of notes

### PR DESCRIPTION
**Pull request by wbamberg**
This PR was inspired by this subthread: https://github.com/mdn/content/pull/4352#discussion_r618824445 and it edits the JS docs to try to use "Note:"  consistently at the start of notes. This should make it easier for a Markdown conversion process to convert these things.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@wbamberg 
I guess it doesn't matter, but the "official" approach is IMO better. That is to make the notes like this:
```
<div class="notecard note">
   <h4>Note</h4>
   <p>blah blah</p>
</div>
```
Why? 
- it is the documented way to do notes in the guide
- notecard is the proper class for an aside. A warning is "notecard warning".
- very easy for a parser to pull out the heading and compare it to the intent.

**Comment by wbamberg** [comment]
FWIW I really don't like using headings for this. I think headings make promises about page structure and hierarchy, and I don't think notes should participate in page structure.

But, that's kind of irrelevant because they're going away anyway. The purpose of this PR is to make our JS notes more consistent, so a conversion process can handle them without a hundred special cases.

There are 231 instances of `div.note` in the JS docs. Of these, the first child node is as follows:

* `STRONG`: 10
* `P`: 215
* `H4`: 2
* `H2`: 1

So if I changed them all to use H2-H6 I'd have to rewrite almost every note.

Of the 215 `P` instances, every one has a `STRONG` first child, and after this PR almost all of those contain "Note:" as the text content. So it would be better to update the 13 other instances, and to make sure all the text contents are consistently "Note:".

**Comment by hamishwillee** [comment]
Sure. If you're just talking about JS and the timeline for conversion is relatively short 100% agree.

**Comment by wbamberg** [comment]
I just pushed a couple more changes, and I think now, all the `div.note` instances in JS are now like:

```
<div class="note" (and sometimes "notecard" too)
  <p><strong>Note:</strong> Yadda yadda</p>
...
</div>
```

File: files/en-us/web/javascript/reference/global_objects/typedarray/values/index.html [hunk 1]
```diff
@@ -56,7 +56,7 @@ <h3 id="Iteration_using_.next">Iteration using .next()</h3>
 iteraror.next().value;         // undefined </pre>
 
 <div class="notecard warning">
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/typedarray/values/index.html:58]: Note or warning?

```suggestion
<div class="notecard note">
```

**Review by Elchi3 — APPROVED** [review]
LGTM! Just one question if it is warning or note

File: files/en-us/web/javascript/reference/global_objects/typedarray/values/index.html (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/javascript/reference/global_objects/typedarray/values/index.html:58]: Oh this is my mistake. That was a warning so I should not have touched it.

File: files/en-us/web/javascript/reference/global_objects/typedarray/values/index.html [hunk 2]
```diff
@@ -56,7 +56,7 @@ <h3 id="Iteration_using_.next">Iteration using .next()</h3>
...
-  <p>One-use: the array iterator object is one use or temporary object</p>
+  <p><strong>Note:</strong> The array iterator object is a one use or temporary object</p>
```
Comment (wbamberg) [inline files/en-us/web/javascript/reference/global_objects/typedarray/values/index.html:59]:
```suggestion
  <p>One-use: the array iterator object is one use or temporary object</p>
```

**Pull request merged**